src/main/java/org/opensearch/knn/training/TrainingJobRunner.java

```diff
@@ -99,12 +96,12 @@ public void execute(TrainingJob trainingJob, ActionListener<IndexResponse> liste
                 // Respond to the request with the initial index response
                 listener.onResponse(indexResponse);
                 train(trainingJob);
-            }, exception -> {
+            }, e -> {
                 // Serialization failed. Let listener handle the exception, but free up resources.
                 jobCount.decrementAndGet();
                 semaphore.release();
-                logger.error("Unable to initialize model serialization: " + exception.getMessage());
-                listener.onFailure(exception);
+                log.error("Unable to initialize model serialization", e);
+                listener.onFailure(e);
```

Comment:
Thought:
I see that you first logged the exception as log.error and then you handled the failure. This is behavior is little strange and here is my thought process: 
listener.onFaiure() function are created to handle the failures/exceptions. It's the responsibility of the failure handler to see if it want to log the exception as error/info/warn + what should be the next step. But with your log.error(), splits the responsibility of the failure handler.

My recommendation would be to remove this log.error before failure handler and lets ensure that our failure handler are handling this exception and as part of failure handling they also see if they want to log the exception or not.

Reply:
That makes sense. Will update.